src/dolphin/unwrap/_unwrap.py

```diff
@@ -304,20 +324,52 @@ def unwrap(
 
         ifg = io.load_gdal(ifg_filename)
         logger.info(f"Goldstein filtering {ifg_filename} -> {filt_ifg_filename}")
-        filt_ifg = goldstein(ifg, alpha=alpha)
+        modified_ifg = goldstein(ifg, alpha=alpha)
         logger.info(f"Writing filtered output to {filt_ifg_filename}")
         io.write_arr(
-            arr=filt_ifg,
+            arr=modified_ifg,
             output_name=filt_ifg_filename,
             like_filename=ifg_filename,
             driver=driver,
             options=opts,
         )
         unwrapper_ifg_filename = filt_ifg_filename
         unwrapper_unw_filename = scratch_unw_filename
-    else:
-        unwrapper_ifg_filename = Path(ifg_filename)
-        unwrapper_unw_filename = Path(unw_filename)
+
+    elif run_interpolation:
+        suf = Path(ifg_filename).suffix
+        if suf == ".tif":
+            driver = "GTiff"
+            opts = list(io.DEFAULT_TIFF_OPTIONS)
+        else:
+            driver = "ENVI"
+            opts = list(io.DEFAULT_ENVI_OPTIONS)
+
+        # temporarily storing the intermediate interpolated rasters in the scratch dir.
+        interp_ifg_filename = (
+            Path(scratchdir or ".")
+            / Path(ifg_filename).with_suffix(".interp" + suf).name
+        )
+        scratch_unw_filename = Path(unw_filename).with_suffix(".interp.unw" + suf)
+
+        ifg = io.load_gdal(ifg_filename)
+        corr = io.load_gdal(corr_filename)
+        logger.info("Masking pixels with correlation above 0.7")
+        coherent_pixel_mask = corr[:] >= 0.7
```

Comment:
I don't think if it should be the responsibility of _unwrap.py to define a mask. I think the mask should be defined upstream and passed to unwrap.